lib/core/cursor.js

```diff
@@ -464,50 +464,41 @@ class CoreCursor extends Readable {
       }
 
       const result = r.message;
-      if (result.queryFailure) {
```

Comment:
I think it's a safe change because according to the [wire protocol docs](https://docs.mongodb.com/manual/reference/mongodb-wire-protocol/), "1 corresponds to QueryFailure. Is set when query failed. Results consist of one document containing an “$err” field describing the failure." So I think we can safely expect the ` Array.isArray(result.documents) && result.documents.length === 1` conditions to be met as well in the case of a `queryFailure`.